torch/csrc/autograd/python_engine.cpp

```diff
@@ -326,8 +330,19 @@ PyTypeObject THPEngineType = {
   THPEngine_new                          /* tp_new */
 };
 
+static void child_atfork() {
+  // Re-initialize the engine in the child process. Since we don't call the
+  // destructor first, this likely leaks some memory.
```

Comment:
Can we try destructing it? `std::thread` should try to do anything funky if we detached the threads.

Reply:
We could and it succeeds in the simple case, but if say a background thread is running at the time of fork it's much more likely to deadlock. I'd rather just leak what's probably a tiny bit of memory in the child process.

Reply:
Maybe this should just set a flag and we can do the re-initialization in `run_backward` / `queue_callback` to avoid non-async-signal-safe functions here.

Reply:
Sounds good